### PR TITLE
Telemetry module verification and release prow jobs

### PR DESCRIPTION
**Pull request by shorim**
**Description**

Changes proposed in this pull request:

- Add a pre-submit job called `verify-telemetry-module` which executes the `run-with-lm` target on every PR.
- Add a post-submit job called `release-telemetry-module` which is triggered when a release tag is pushed. It executes the `release` target.

**Related issue(s)**

kyma-project/telemetry-manager#113

### CODE REVIEW

**Comment by dekiel** [comment]
This PR is not ready for review. Removing myself.

File: templates/data/telemetry-manager-data.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+templates:
+  - from: generic.tmpl
+    render:
+      - to: ../../prow/jobs/telemetry-manager/telemetry-manager-generic.yaml
```
Comment (Sawthis) [inline templates/data/telemetry-manager-data.yaml:4]:
```suggestion
      - to: ../../prow/jobs/telemetry-manager/telemetry-manager-generic-new.yaml
```

**Comment by Sawthis** [comment]
@shorim is this PR ready for review?

**Comment by chrkl** [comment]
> @shorim is this PR ready for review?

Still needs some adjustments. Unfortunately, @shorim is off this week and I cannot make it a draft.

**Review by rakesh-garimella — APPROVED** [review]

**Review by the1bit — APPROVED** [review]
Partially tested. For the release-telemetry-module , it was not possible to fully test it, because I need to push the module image to prod registry and that is not possible on an untrusted cluster which is used by pjtester

**Pull request merged**